src/core/common/timer.hpp

```diff
@@ -181,6 +181,47 @@ class TimerMilli : public Timer
      */
     void Stop(void);
 
+    /**
+     * This static method returns the time diff in milliseconds between @p aTime1 and @p aTime2.
+     *
+     * @param[in]   aStart  The start time.
+     * @param[in]   aEnd    The end time.
+     *
+     * @returns The time diff in milliseconds.
+     *
+     */
+    static int32_t Diff(uint32_t aStart, uint32_t aEnd) { return static_cast<int32_t>(aEnd - aStart); }
+
+    /**
+     * This static method returns the time elapsed in milliseconds from @p aStart to @p aEnd.
+     *
+     * @note This method asserts @p aEnd is after @p aTime.
+     *
+     * @param[in]   aStart  The start time.
+     * @param[in]   aEnd    The end time.
+     *
+     * @returns The elapsed time in milliseconds.
+     *
+     */
+    static uint32_t Elapsed(uint32_t aStart, uint32_t aEnd)
+    {
+        int32_t elapsed = Diff(aStart, aEnd);
+        assert(elapsed >= 0);
```

Comment:
@bukepo, can you please help describe your intention for the assert check here? 
Is it to not allow difference of larger than `2^31 - 1` ticks when comparing two times (if so why)? 

Note that if we know that `aStart` is before `aEnd` then the `aEnd - aStart` (as a `uint32_t`) will always give us the 
correct time difference (no matter the wrapping of the time values and even if the `aEnd` is `2^32-1` ticks after of `aStart`).

Reply:
@abtink This method should only be called when `aStart` is before `aEnd`. This assert prevents possible invalid usage.